### PR TITLE
Add Java 17 to the build/test matrix

### PR DESCRIPTION
**Pull request by cljohnso**
Java 8 continues to be the distribution version for this component ... but Java 17 is added to the test matrix.

### CODE REVIEW

**Comment by cljohnso** [comment]
@akomakom The number of tasks run for this smells like a little overkill but we really want this to be tested under 17.  We can have a discussion about a way to pare this down.

**Comment by cljohnso** [comment]
Looks like I have some deprecations I need to address for building under Java 17.  Right now, compilation under Java 17 isn't a strict requirement but it should compile cleanly none the less.

**Comment by cljohnso** [comment]
Looks like there's a file system timestamp resolution difference in Java 17 ... attempting to isolate and address.

**Comment by cljohnso** [comment]
I re-ran the ubuntu 17 job until it failed ... seems there's an issue with coordination in test `InterprocessCyclicBarrierTest.testTwoThreadsCommonBarrierInterruptedBeforeRegister` -- noodling on whether or not I have enough information to address it now or if I need to add some additional diagnostic output.  (I _think_ there's a race between the two barrier participant threads and I need to add an additional alignment point.)

File: tools/src/test/java/org/terracotta/utilities/concurrent/InterprocessCyclicBarrierTest.java [hunk 1]
```diff
@@ -16,15 +16,24 @@
 package org.terracotta.utilities.concurrent;
 
 
+import org.junit.AfterClass;
```
Comment (cljohnso) [inline tools/src/test/java/org/terracotta/utilities/concurrent/InterprocessCyclicBarrierTest.java:19]: Changes in this file are best evaluated when ignoring whitespace changes.

File: tools/src/test/java/org/terracotta/utilities/concurrent/InterprocessCyclicBarrierTest.java [hunk 2]
```diff
@@ -55,6 +64,49 @@ public class InterprocessCyclicBarrierTest  {
   @Rule
   public final TemporaryFolder temporaryFolder = new TemporaryFolder();
 
+  @Rule
+  public final Timeout timeout = Timeout.seconds(10L);
+
+  @Rule
+  public final TestWatcher watcher = new TestWatcher() {
```
Comment (cljohnso) [inline tools/src/test/java/org/terracotta/utilities/concurrent/InterprocessCyclicBarrierTest.java:71]: This was added to help identify the hanging test -- under normal conditions, these tests produce no output making a stalled test impossible to identify.

**Comment by akomakom** [comment]
> @akomakom The number of tasks run for this smells like a little overkill but we really want this to be tested under 17. We can have a discussion about a way to pare this down.

If each matrix combination takes a couple of minutes then I don't see a big issue with running them all.

**Review by GaryWKeim — APPROVED** [review]

**Review by chrisdennis — APPROVED** [review]

**Pull request merged**